### PR TITLE
windows set ime area not work

### PR DESCRIPTION
**Pull request by wjian23**
I encounter  the Api of  set_ime_cursor_area not work on windows with Sogou Ime when i develop a new rust GUI framework Ribir(https://github.com/RibirX/Ribir). I try to fix it here, please take a look。

### CODE REVIEW

File: src/platform_impl/windows/ime.rs [hunk 1]
```diff
@@ -124,16 +124,24 @@ impl ImeContext {
...
+            bottom: y + height,
         };
         let candidate_form = CANDIDATEFORM {
             dwIndex: 0,
             dwStyle: CFS_EXCLUDE,
             ptCurrentPos: POINT { x, y },
             rcArea: rc_area,
         };
+        let composition_form = COMPOSITIONFORM {
+            dwStyle: CFS_EXCLUDE,
```
Comment (kchibisov) [inline src/platform_impl/windows/ime.rs:136]: There's no such option for the form.

**Review by kchibisov — CHANGES_REQUESTED** [review]
Sounds like IME is just bugged, from what I can read on https://learn.microsoft.com/en-us/windows/win32/api/imm/ns-imm-candidateform `rc_area` must be excluded, what you've changed uses wrong style and solely about positioning, but not about exclusion.

Maybe the rectangle passed is not correct, but `COMPOSITION_FORM` shouldn't be required. I'd also suggest to test other IMEs.

File: src/platform_impl/windows/ime.rs (hunk 1 above)
Comment (wjian23) [reply src/platform_impl/windows/ime.rs:136]: yeah according to the official document should use CFS_POINT here. i'll take a fix

File: src/platform_impl/windows/ime.rs [hunk 2]
```diff
@@ -124,16 +124,24 @@ impl ImeContext {
...
+        let composition_form = COMPOSITIONFORM {
+            dwStyle: CFS_POINT,
+            ptCurrentPos: POINT { x, y: y + height },
+            rcArea: rc_area,
+        };
 
-        unsafe { ImmSetCandidateWindow(self.himc, &candidate_form) };
+        unsafe {
+            ImmSetCompositionWindow(self.himc, &composition_form);
+            ImmSetCandidateWindow(self.himc, &candidate_form);
```
Comment (kchibisov) [inline src/platform_impl/windows/ime.rs:143]: Why the order is like that and not other way around? Could we set after creating data for the request we want to perform and not doing just set-all in the end of function?

Also, if you want me to re-review you should request it.

Comment (wjian23) [reply src/platform_impl/windows/ime.rs:143]: @kchibisov the order doesn't matter here. Create and set will be more nature, i have modify based on your comments

**Review by kchibisov — APPROVED** [review]

**Comment by kchibisov** [comment]
Thanks!

**Pull request merged**